test_vm/src/lib.rs

```diff
@@ -872,15 +907,19 @@ impl<'invocation, 'bs> Runtime for InvocationCtx<'invocation, 'bs> {
             caller_validated: false,
             policy: self.policy,
             subinvocations: RefCell::new(vec![]),
+            actor_exit: RefCell::new(None),
         };
-        let res = new_ctx.invoke();
-
+        let res = new_ctx.invoke_actor();
         let invoc = new_ctx.gather_trace(res.clone());
         RefMut::map(self.subinvocations.borrow_mut(), |subinvocs| {
             subinvocs.push(invoc);
             subinvocs
         });
-        res
+
+        Ok(Response {
+            exit_code: res.as_ref().err().map(|e| e.exit_code()).unwrap_or(ExitCode::OK),
```

Comment:
Just `unwrap()`? If it's not present, seems like a bug somewhere (probably in this test VM).

Reply:
The unwrap_or is for the case when `res` is `Ok()` -- in that case `err()` returns `None`. The multi-line, but clearer way to write this is:

```
        Ok(Response {
            exit_code: match res.as_ref() {
                Ok(_) => ExitCode::OK,
                Err(e) => e.exit_code(),
            },
```